### ISSUE TITLE
Using AppendPathSegment("/") creates a URL which ends in "//"

### ISSUE THREAD

**Issue by Eran-BenShahar** [tmenier/Flurl#609]
```
var uri = new Uri("https://www.site.com/a/b/c");
uri.AppendPathSegment("/").ToString()
```
Will result in:
"https://a.b.com/a/b/c//"
Expected behavior:
"https://a.b.com/a/b/c/" or "https://a.b.com/a/b/c"
<br>
Also, if we add a non-root path it works as expected.
```
var uri = new Uri("https://www.site.com/a/b/c");
uri.AppendPathSegment("/d").ToString()
```
Results in "https://a.b.com/a/b/c/d"

**Comment by tmenier** [comment]
Thanks for reporting, confirmed and fixed. Will be released in 3.0.2.

**Comment by tmenier** [comment]
Published to NuGet in Flurl 3.0.2
